src/CommunityToolkit.Maui.BindablePropertySG/BindablePropertySG.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
+using System.Text;
+using CommunityToolkit.Maui.BindablePropertySG.Helpers;
+using CommunityToolkit.Maui.BindablePropertySG.Models;
+using CommunityToolkit.Maui.SourceGenerators.Helpers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CommunityToolkit.Maui.BindablePropertySG;
+
+[Generator]
+public class BindablePropertySG : IIncrementalGenerator
+{
+	static readonly SemanticValues emptySemanticValues = new(default, Array.Empty<BPInfo>().ToImmutableArray());
+
+	const string bpFullName = "global::Microsoft.Maui.Controls.BindableProperty";
+	const string bindingModeFullName = "global::Microsoft.Maui.Controls.";
+
+	const string bpAttribute = """
+#nullable enable
+namespace CommunityToolkit.Maui.BindablePropertySG;
+
+[AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
+public sealed class BindablePropertyAttribute<TReturnType> : Attribute
+{
+	public string PropertyName { get; } = string.Empty;
+	public Type? OwnerType { get; set; }
+	public object? DefaultValue { get; set; }
+	public string DefaultBindingMode { get; set; } = string.Empty;
+	public string ValidateValueMethodName { get; set; } = string.Empty;
+	public string PropertyChangedMethodName { get; set; } = string.Empty;
+	public string PropertyChangingMethodName { get; set; } = string.Empty;
+	public string CoerceValueMethodName { get; set; } = string.Empty;
+	public string DefaultValueCreatorMethodName { get; set; } = string.Empty;
+
+	public BindablePropertyAttribute(string propertyName)
+	{
+		PropertyName = propertyName;
+	}
+}
+""";
+
+	public void Initialize(IncrementalGeneratorInitializationContext context)
+	{
+		context.RegisterPostInitializationOutput(ctx => ctx.AddSource("BindablePropertyAttribute.g.cs", SourceText.From(bpAttribute, Encoding.UTF8)));
+
+		var provider = context.SyntaxProvider.ForAttributeWithMetadataName("CommunityToolkit.Maui.BindablePropertySG.BindablePropertyAttribute`1",
+			SyntaxPredicate, SemanticTransform)
+			.Where(static x => x.ClassInformation != default || !x.BPInfos.IsEmpty)
+			.Collect()
+			.SelectMany(static (types, _) => types);
```

Comment:
This was necessary to avoid the provider to be a collection. I prefer to do it here instead of having a `foreach` loop inside the `Execute` method